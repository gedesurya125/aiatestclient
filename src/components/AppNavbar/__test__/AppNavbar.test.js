import React from 'react';
import reactDOM from 'react-dom';
import {render, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import AppNavBar from '../index';
import { Provider } from 'react-redux';
import store from '../../../redux/store';


const AppNavBarWithRedux = () => <Provider store={store}><div data-testid="app-navbar"><AppNavBar/></div></Provider>

it('component should render without crash', () => {
  const div = document.createElement('div');
  reactDOM.render(<AppNavBarWithRedux/>, div);
  reactDOM.unmountComponentAtNode(div);
})

it('component should render correctly', () => {
  const {getByTestId} = render(<AppNavBarWithRedux/>)
  const appNavbar = getByTestId('app-navbar');
  const searchInputContainer = getByTestId('search-input-container');
  const logoSection = getByTestId('logo-section');
  expect(appNavbar).toContainElement(searchInputContainer);
  expect(appNavbar).toContainElement(logoSection);
})

it('Match snapshot 1', () => {
  const tree = renderer.create(<AppNavBarWithRedux/>);
  expect(tree).toMatchSnapshot();
})
