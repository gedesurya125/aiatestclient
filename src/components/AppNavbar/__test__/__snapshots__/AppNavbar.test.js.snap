// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot 1 1`] = `
<div
  data-testid="app-navbar"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1carciz-MuiPaper-root-MuiAppBar-root"
    data-testid="hide-on-scroll"
    style={
      Object {
        "visibility": undefined,
      }
    }
  >
    <div
      className="css-1yd8edg"
      data-testid="logo-section"
    >
      <img
        alt="..."
        data-testid="image"
        src="logoWhite.svg"
      />
    </div>
    <div
      className="css-1pmg02h"
      data-testid="search-input-container"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-8jzbc3-MuiFormControl-root-MuiTextField-root"
        data-testid="search-input"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
          >
            <span
              className="notranslate"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-s040oo-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart css-1ua80n0-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search by tag name"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1a1ihhj"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Search
      </button>
    </div>
  </header>
</div>
`;
